synapse/http/matrixfederationclient.py

```diff
@@ -629,53 +639,52 @@ def build_auth_headers(
 
     async def put_json(
         self,
-        destination,
-        path,
-        args={},
-        data={},
-        json_data_callback=None,
-        long_retries=False,
-        timeout=None,
-        ignore_backoff=False,
-        backoff_on_404=False,
-        try_trailing_slash_on_400=False,
-    ):
+        destination: str,
+        path: str,
+        args: Optional[QueryArgs] = None,
+        data: Optional[JsonDict] = None,
+        json_data_callback: Optional[Callable[[], JsonDict]] = None,
+        long_retries: bool = False,
+        timeout: Optional[int] = None,
+        ignore_backoff: bool = False,
+        backoff_on_404: bool = False,
+        try_trailing_slash_on_400: bool = False,
+    ) -> Union[JsonDict, list]:
```

Comment:
Are there any server-server API endpoints that ever return a list rather than a json object? (I've come across a bunch of typing complaints that would vanish if this were simply `JsonDict`, and I'm wondering if that change is legitimate.)

Reply:
A bunch return `[int, {...}]` where the int is also the status code of the response. Don't ask why.